flink-yarn/src/test/java/org/apache/flink/yarn/ContainerRequestReflectorTest.java

```diff
@@ -0,0 +1,171 @@
+import java.lang.reflect.Method;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+
+/** Tests for {@link ContainerRequestReflector}. */
+public class ContainerRequestReflectorTest extends TestLogger {
+
+    @Test
+    public void testGetContainerRequestIfConstructorPresent() {
+        final ContainerRequestReflector containerRequestReflector =
+                new ContainerRequestReflector(ContainerRequestWithConstructor.class);
+        Resource resource = Resource.newInstance(100, 1);
+        Priority priority = Priority.newInstance(1);
+
+        AMRMClient.ContainerRequest containerRequest =
+                containerRequestReflector.getContainerRequest(resource, priority, "GPU");
+        assertTrue(containerRequest instanceof ContainerRequestWithConstructor);
+        ContainerRequestWithConstructor containerRequestWithConstructor =
+                (ContainerRequestWithConstructor) containerRequest;
+        assertEquals("GPU", containerRequestWithConstructor.getNodeLabelsExpression());
+
+        containerRequest = containerRequestReflector.getContainerRequest(resource, priority, null);
+        assertTrue(containerRequest instanceof AMRMClient.ContainerRequest);
+        assertFalse(containerRequest instanceof ContainerRequestWithConstructor);
+
+        containerRequest = containerRequestReflector.getContainerRequest(resource, priority, "");
+        assertTrue(containerRequest instanceof AMRMClient.ContainerRequest);
+        assertFalse(containerRequest instanceof ContainerRequestWithConstructor);
+    }
+
+    @Test
+    public void testGetContainerRequestIfConstructorAbsent() {
+        final ContainerRequestReflector containerRequestReflector =
+                new ContainerRequestReflector(ContainerRequestWithoutConstructor.class);
+        Resource resource = Resource.newInstance(100, 1);
+        Priority priority = Priority.newInstance(1);
+
+        AMRMClient.ContainerRequest containerRequest =
+                containerRequestReflector.getContainerRequest(resource, priority, "GPU");
+        assertTrue(containerRequest instanceof AMRMClient.ContainerRequest);
+        assertFalse(containerRequest instanceof ContainerRequestWithConstructor);
+
+        containerRequest = containerRequestReflector.getContainerRequest(resource, priority, null);
+        assertTrue(containerRequest instanceof AMRMClient.ContainerRequest);
+        assertFalse(containerRequest instanceof ContainerRequestWithConstructor);
+
+        containerRequest = containerRequestReflector.getContainerRequest(resource, priority, "");
+        assertTrue(containerRequest instanceof AMRMClient.ContainerRequest);
+        assertFalse(containerRequest instanceof ContainerRequestWithConstructor);
+    }
+
+    @Test
+    public void testGetContainerRequestWithoutYarnSupport() {
```

Comment:
I think we cannot verify anything in this test. So, just creating three `ContainerRequest` without any exception would be good.